micropip/package_index.py

```diff
@@ -2,17 +2,19 @@
 import string
 import sys
 from collections import defaultdict
-from collections.abc import Generator
+from collections.abc import Callable, Generator
 from dataclasses import dataclass
+from functools import partial
 from typing import Any
 
 from packaging.utils import InvalidWheelFilename
 from packaging.version import InvalidVersion, Version
 
-from ._compat import fetch_string
+from ._compat import fetch_string_and_headers
 from ._utils import is_package_compatible, parse_version
+from .externals.mousebender.simple import from_project_details_html
 
-DEFAULT_INDEX_URLS = ["https://pypi.org/pypi/{package_name}/json"]
+DEFAULT_INDEX_URLS = ["https://pypi.org/simple"]
```

Comment:
Can you confirm we are never actually downloading the top level index here, because it's >3MB compressed in the case of PyPI.

Reply:
Yes, I can confirm that we are not downloading the top level index, unless someone changes the index URL to `https://pypi.org` and do `micropip.install("simple")` :)